packages/marketplace/contracts/TransferManager.sol

```diff
@@ -265,28 +260,14 @@ abstract contract TransferManager is Initializable, ITransferManager {
     ) internal returns (uint256 remainder) {
         remainder = paymentSide.asset.value;
         uint256 feePrimary = protocolFeePrimary;
-        uint256 feeSecondary = protocolFeeSecondary;
         LibAsset.Bundle memory bundle = LibAsset.decodeBundle(nftSide.asset.assetType);
 
-        remainder = _processERC721Bundles(
-            paymentSide,
-            nftSide,
-            nftSideRecipient,
-            remainder,
-            feePrimary,
-            feeSecondary,
-            bundle
-        );
-        remainder = _processERC1155Bundles(
-            paymentSide,
-            nftSide,
-            nftSideRecipient,
-            remainder,
-            feePrimary,
-            feeSecondary,
-            bundle
-        );
-        remainder = _processQuadBundles(paymentSide, nftSideRecipient, remainder, feeSecondary, bundle);
+        remainder = _processERC721Bundles(paymentSide, nftSide, nftSideRecipient, remainder, feePrimary, bundle);
+        remainder = _processERC1155Bundles(paymentSide, nftSide, nftSideRecipient, remainder, feePrimary, bundle);
+        uint256 quadSize = bundle.quads.xs.length;
+        if (quadSize > 0 && _isTSBBundleSeller(nftSide.account)) {
+            remainder = _processQuadBundles(paymentSide, nftSideRecipient, remainder, feePrimary, quadSize, bundle);
+        }
```

Comment:
We dont need this check for isTSBBundleSeller here, users should not be able to bundle quads because it should fail on `_isPrimaryMarket` check

Reply:
`_isPrimaryMarket` isn't checked for bundled quad. 
and we need only TSB_SECONDARY_MARKET_SELLER to be checked for royalties to be paid for quad.